### ISSUE TITLE
Adding a new crop to existing role causes issues

### ISSUE THREAD

**Issue by IllyaMoskvin** [area17/twill#1193]
## Description

I'm attempting to add a new crop to an existing image role.

https://twill.io/docs/#models

This is causing an error on the frontend in the CMS.

When I try to edit an _existing_ media and select the new crop in the crop modal, I get an undefined error here:

https://github.com/area17/twill/blob/377e5e0b27916861caa448ef899ea0e3fbeff648/frontend/js/components/Cropper.vue#L157

### Steps to reproduce

1. Add a media definition like this to some model:
    ```
    public $mediasParams = [
        'hero' => [
            'default' => [
                [
                    'name' => 'default',
                    'ratio' => 16 / 9,
                ],
            ],
        ],
    ];
    ```
2. Create a new instance of that model via the CMS, fill in the media, set a crop, and save it.
3. Add a new crop:
    ```
    public $mediasParams = [
        'hero' => [
            'default' => [
                [
                    'name' => 'default',
                    'ratio' => 16 / 9,
                ],
            ],
            'mobile' => [
                [
                    'name' => 'mobile',
                    'ratio' => 9 / 16,
                ],
            ],
        ],
    ];
    ```
4. Open that media in the CMS for cropping, click on the new crop.
5. You'll see the following error in the console:
    ```
    TypeError: Cannot read properties of undefined (reading 'name')
    ```

This is happening because `this.crop` resolves to `undefined`.

### Expected result

Adding a new crop to an existing role does not cause an error.

### Actual result

As described above.

### Versions

**Twill:** 2.4.0
**Laravel:** 6.20.34
**PHP:** 7.2
**DB:** Postgres 13

**Comment by ifox** [comment]
Hi guys,

The inability of Twill to pick up changes to the crops configuration, forcing users to remove the associated image to allow a new set of crops, is a known issue that we should prioritize. It can be a critical issue if a crop position has been manually tailored by an editor.

I am intrigued by the difference of behaviors. What @pboivin experiences is what I'm used to, not the error in console. We'll investigate.

Thanks for reporting @IllyaMoskvin!

**Comment by mattrabe** [comment]
I am also seeing the error in the console, and can confirm that this is related to **_changing_** the list of `mediaParams` after saving a crop. e.g. if you have "set A" of mediaParams, save crops on an image, then change to "set B" of mediaParams and attempt to crop that same image, the error occurs. In my case, `TypeError: Cannot read properties of undefined (reading 'name')` plus the fact that the crop window will not match the ratio spec'd in your mediaParams.

It would appear, then, that crop information is likely stored to the database in a rigid manner, wherein if the set of mediaParams changes later, things break.

Agree with this being a high priority fix needed.

**Comment by pboivin** [comment]
Hi @mattrabe, have a look at the 2 PRs linked above: #1258 and #1289. This will be released soon with Twill 2.6 but is already available in the `2.x` branch if you want to test it out.

**Comment by mattrabe** [comment]
thanks @pboivin!
